src/node/https.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { urlToHttpOptions, isURL } from 'node-internal:internal_url';
+import { ClientRequest } from 'node-internal:internal_http_client';
+import { Agent, globalAgent } from 'node-internal:internal_https_agent';
+import type { IncomingMessageCallback } from 'node-internal:internal_http_util';
+import type { RequestOptions } from 'node:http';
+
+export function request(...args: unknown[]): ClientRequest {
```

Comment:
Nit: should probably document the actual non-generic signatures as function decls above this.